NYPL-discovery/discovery-front-end
# Closure scenarios

**What's this do?**

Adds the `closedLocations` env parameter back in, which got removed.
Adds some more flexibility in the way we display request options to account for more possible closure scenarios. In particular:

- edd is now a possible location so we can shut down edd independently
- recap and non-recap can have separate closed locations (including edd).

By file:

- README: add some documentation about values for the `closedLocations` parameter
- HoldRequest: depending on whether the item `isRecap`, we add the recap or non-recap specific locations to the `closedLocations` to get an item-specific list of closed locations (`itemClosedLocations`). Then we filter out any of these from the `deliveryLocations` prop, and also check if `edd` is closed, before displaying options
- ItemTableRow: add in the recap or non-recap specific closed locations before checking whether to display the request button
- The remaining changes are just to add the relevant variables in `appConfig` and `webpac` and add some tests
 
**Why are we doing this? (w/ JIRA link if applicable)**
This is `SCC-2425`, also relevant is `SCC-2394`

**Do these changes have automated tests?**
Yes, we are checking that the `HoldRequest` page can close edd independently of locations, and can close recap and non-recap independently. Also check that the `ItemTableRow` can turn off requests for recap and non-recap independently.

**How should this be QAed?**
Set the `CLOSED_LOCATIONS`, `RECAP_CLOSED_LOCATIONS`, and `NON_RECAP_CLOSED_LOCATIONS` to different values and check that the search results page and bib page suppress the request button for recap/non-recap items when everything/recap/non-recap is all closed. Check that the HoldRequest page turns off edd/physical locations for all/recap/non-recap locations.

**Dependencies for merging? Releasing to production?**

**Has the application documentation been updated for these changes?**
Yes

**Did someone actually run this code to verify it works?**
PR author tested locally.
